### ISSUE TITLE
you have to have GooglePlay account

### ISSUE THREAD

**Issue by damko** [mysteriumnetwork/mysterium-vpn-mobile#423]
Hello, I run lineageos on my google free android. I also run a mysterium node because I love the idea of community driven VPN network.

I upgraded to version 1.0.4 and even 1.0.3 and to my IMMENSE surprise they both crash as soon as started because "you have to have Google Play account"

I tried to hide magisk but there is no difference.

My questions:
* what's the latest version I can use without the Google Play requirement?
* are you considering to keep Google Play as a requirement for your VPN client?

**Comment by tadaskay** [comment]
I can answer your second question straight away - no, we are aiming to get rid of this. However, per updated google play policies, they support only google payments which we will be looking into implementing soon for the wallet topups. That being said, I don't see a way how you could top up your account without using a Google play account. A workaround would be to top it up on the desktop client and import the same identity into the mobile app, but that's far too inconvenient.

Regarding crashes, @ArtemHryhorovGeniusee could you test and fix such case?

**Comment by ArtemHryhorovGeniusee** [comment]
In last version - 1.0.4. there is no toast with  "you have to have Google Play account". The only toast which exist - about permission. I will re-check the GooglePlay check flow and make update, but make sure you are using last version

**Comment by damko** [comment]
I'm sure I've downloaded  1.0.3 APK from releases here on github. The 1.0.4 came straight from the website. They both prompted "you have to have a Google Play account" otherwise I could I report such a message if I didn't see it?

Anyway, given the doubt, in the next hours I'll grab the latest from releases and let you know.

**Comment by damko** [comment]
I've downloaded 1.0.4 from releases. It crashes at start but it does not show the bubble "you have to have Google Play account". However the way it crashes feels suspiciously identical to 1.0.3 i.e. right after the logo appears over the purple background (except for the bubble).

I installed scoop and this is its log:

    FATAL EXCEPTION: main
    Process: network.mysterium.vpn, PID: 13594
    com.google.android.play.core.tasks.RuntimeExecutionException: com.google.android.play.core.install.InstallException: Install Error(-9): The Play Store app is either not installed or not the official version. (https://developer.android.com/reference/com/google/android/play/core/install/model/InstallErrorCode#ERROR_PLAY_STORE_NOT_FOUND)
        at com.google.android.play.core.tasks.m.getResult(Unknown Source:16)
        at updated.mysterium.vpn.ui.splash.SplashActivity$checkForGoogleMarketUpdates$1.onComplete(SplashActivity.kt:130)
        at com.google.android.play.core.tasks.a.run(Unknown Source:23)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)
    Caused by: com.google.android.play.core.install.InstallException: Install Error(-9): The Play Store app is either not installed or not the official version. (https://developer.android.com/reference/com/google/android/play/core/install/model/InstallErrorCode#ERROR_PLAY_STORE_NOT_FOUND)
        at com.google.android.play.core.appupdate.o.i(Unknown Source:21)
        at com.google.android.play.core.appupdate.o.a(Unknown Source:4)
        at com.google.android.play.core.appupdate.d.getAppUpdateInfo(Unknown Source:8)
        at updated.mysterium.vpn.ui.splash.SplashActivity.checkForGoogleMarketUpdates(SplashActivity.kt:129)
        at updated.mysterium.vpn.ui.splash.SplashActivity.ensureVpnServicePermission(SplashActivity.kt:195)
        at updated.mysterium.vpn.ui.splash.SplashActivity.onCreate(SplashActivity.kt:63)
        at android.app.Activity.performCreate(Activity.java:7824)
        at android.app.Activity.performCreate(Activity.java:7813)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1306)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3245)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        ... 5 more
        
So you are right when you say that there is no toast but the Google Play constraint is still there: _The Play Store app is either not installed or not the official version_

Basically my initials questions are still standing:
* what's the latest version I can use without the Google Play requirement?
* are you considering to keep Google Play as a requirement for your VPN client?

**Comment by ArtemHryhorovGeniusee** [comment]
what's the latest version I can use without the Google Play requirement? -> Try to use new  1.0.5 Tag release

**Comment by damko** [comment]
@ArtemHryhorovGeniusee I tested 1.0.5 and it works. Thank you!
